formatTest/unit_tests/expected_output/infix.re

```diff
@@ -348,3 +368,586 @@ let myFunc
+     */
+    x.contents =
+      something ?
+        (x.contents = somethingElse: x) : z;
+    y = something ? (y = somethingElse: x) : z;
+    arr.(0) =
+      something ?
+        (arr.(0) = somethingElse: x) : z;
+    bigArr.{0} =
+      something ?
+        (bigArr.{0} = somethingElse: x) : z;
+    str.[0] =
+      something ?
+        (str.[0] = somethingElse: x) : z;
+
+    /**
+     * Various precedence groupings.
+     */
+    true ? true ? false : false : false;
+    /* Is the same as */
+    true ? true ? false : false : false;
+    /*
+     * Just some examples of how prefix will be printed.
+     */
+    - x + (something.contents = y);
+    - x + (something = y);
+    - x + something.contents := y;
+    - x + something := y;
+    x + (- (something.contents = y));
+    x + (- (something = y));
+    x + (- something.contents) := y;
+    x + (- something) := y;
+    x.contents || something + 1 ?
+      - hello : goodbye;
+    bigArr.{0} || - something + 1 ?
+      hello : goodbye;
+    let result = - x + (something.contents = y);
+    /* Prefix minus is actually sugar for regular function identifier ~-*/
+    let result = 2 + (- add 4 0);
+    /* Same as */
+    let result = 2 + (- add) 4 0;
+    /* Same as */
+    let result = 2 + (- add 4 0);
+    /* That same example but with ppx attributes on the add application */
+    let result = 2 + (- add 4 0 [@ppx]);
+    /* Same as */
+    let result = 2 + (- add) 4 0 [@ppx];
+    /* Same as */
+    let result = 2 + (- add 4 0 [@ppx]);
+    /* Multiple nested prefixes */
+    let result = 2 + (- (- (- add 4 0)));
+    /* And with attributes */
+    let result =
+      2 + (
+        - (- (- add 4 0 [@onAddApplication]))
+      );
+
+    /**
+     * TODO: Move all of these test cases to attributes.re.
+     */
+    /* Attribute on the prefix application */
+    let res = (- something blah blah) [@attr];
+    /* Attribute on the regular function application, not prefix */
+    let res = - something blah blah [@attr];
+    let attrOnPrefix = (-1) [@ppxOnPrefixApp];
+    let attrOnPrefix = 5 + (-1);
+    let result =
+      arr.[0] [@ppxAttributeOnSugarGetter];
+
+    /**
+     * Unary plus/minus has lower precedence than prefix operators:
+     * And unary plus has same precedence as unary minus.
+     */
+    let res = - !record;
+    /* Should be parsed as: */
+    let res = - !record;
+    /* Although that precedence ranking doesn't likely have any effect in that
+     * case. */
+
+    /**
+     * And this
+     */
+    let res = - (+ callThisFunc ());
+    /* should be parsed as: */
+    let res = - (+ callThisFunc ());
+
+    /**
+     * And this
+     */
+    let res = !(- callThisFunc ());
+    /* Should be parsed (and should remain printed as: */
+    let res = !(- callThisFunc ());
+    let res = !x [@onApplication];
+    let res = !(x [@onX]);
+    let res = !(x [@onX]);
+    (something.contents = "newvalue")
+    [@shouldBeRenderedOnEntireSetField];
```

Comment:
This is also another fix/improvement that the refactor made natural/easy. Before we were dropping attributes on some expressions such as `x.y = foo`.